pkg/certificate/certificate.go

```diff
@@ -264,18 +264,37 @@ func InitKubeconfig(
 		return err
 	}
 
-	// expose the controlplane kubeconfig in a secret
+	// expose controlplane in-cluster kubeconfig in a secret
+	if err := util.KubeconfigWroteToSecret(
```

Comment:
who will use the in-cluster kubeconfig? is it for hosted mode case?

Reply:
yes

Reply:
Got it. you need rebase this PR.

Reply:
done